File: unity/oelMVCS/Assets/XTC/oelMVCS/Sample/UIFacade.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace XTC.oelMVCS
{
    public class UIFacade : MonoBehaviour
    {
        public static Dictionary<string, UIFacade> uiFacades = new Dictionary<string, UIFacade>();
        public static Logger logger
        {
            get;
            set;
        }

        public string UUID = "";

        public static UIFacade Find(string _uuid)
        {
            if (uiFacades.ContainsKey(_uuid))
                return uiFacades[_uuid];
            return null;
        }

        public void Register()
        {
            logger.Info("register facade {0}", UUID);
            if (uiFacades.ContainsKey(UUID))
                throw new System.ArgumentException("facade {0} exists", UUID);
            else
                uiFacades[UUID] = this;
        }

        public void Cancel()
        {
            logger.Info("cancel facade {0}", UUID);
            if (uiFacades.ContainsKey(UUID))
                uiFacades.Remove(UUID);
        }
    }
}//namespace

File: python/oelMVCS/UIFacade.py
# -*- coding: utf-8 -*-

class UIFacade:

    facades = {}

    def Register(_uuid, _facade):
        if _uuid in UIFacade.facades:
            raise KeyError('facade is exists')
        UIFacade.facades[_uuid] = _facade

    def Cancel(_uuid):
        if not _uuid in UIFacade.facades:
            raise KeyError('facade not found')
        del UIFacade.facades[_uuid] 

    def Find(_uuid):
        return UIFacade.facades[_uuid]
